backend/metadata_writer/src/metadata_writer.py

```diff
@@ -137,13 +137,21 @@ def is_kfp_v2_pod(pod) -> bool:
 
 while True:
     print("Start watching Kubernetes Pods created by Argo")
-    for event in k8s_watch.stream(
+    pod_stream = k8s_watch.stream(
         k8s_api.list_namespaced_pod,
         namespace=namespace_to_watch,
         label_selector=ARGO_WORKFLOW_LABEL_KEY,
         timeout_seconds=1800,  # Sometimes watch gets stuck
         _request_timeout=2000,  # Sometimes HTTP GET gets stuck
-    ):
+    )
+    if not namespace_to_watch:
```

Comment:
the first watch is still called when namespace_to_watch is empty

Can there be side effects? Shall we put both branches into an if else?

Reply:
You are correct! I will push a fix immediately.